json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+class WPCOM_JSON_API_Get_Comments_Tree_Endpoint extends WPCOM_JSON_API_Endpoint {
+	/**
+	 * Retrieves a list of comment data for a given site.
+	 *
+	 * @param   {string} $status   filter by status: 'all' | 'approved' | 'pending' | 'spam' | 'trash'
+	 * @param   {int}    $start_at first comment to search from going back in time
+	 * @returns {array}            list of comment information matching query
+	 */
+	public function get_site_tree( $status, $start_at = PHP_INT_MAX ) {
+		global $wpdb;
+		$max_comment_count = 10000;
+		$total_count = $this->get_site_tree_total_count( $status );
+		$db_status = $this->get_comment_db_status( $status );
+
+		$db_comments = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT comment_ID, comment_post_ID, comment_parent " .
+				"FROM $wpdb->comments AS comments " .
+				"INNER JOIN $wpdb->posts AS posts ON comments.comment_post_ID = posts.ID " .
+				"WHERE comment_type = '' AND comment_ID <= %d AND ( %s = 'all' OR comment_approved = %s ) " .
+				"ORDER BY comment_ID DESC " .
+				"LIMIT %d",
+				(int) $start_at, $db_status, $db_status, $max_comment_count
+			),
+			ARRAY_N
+		);
+
+		return array(
+			$total_count,
+			array_map( function( $comment ) {
```

Comment:
Jetpack needs to be compatible with PHP 5.2, so this will not work, unfortunately. This is why Travis is breaking.